packages/server/src/middleware/triggerRowActionAuthorised.ts

```diff
@@ -1,40 +1,52 @@
 import { Next } from "koa"
-import { Ctx } from "@budibase/types"
-import { paramSubResource } from "./resourceId"
+import { PermissionLevel, PermissionType, UserCtx } from "@budibase/types"
 import { docIds } from "@budibase/backend-core"
 import * as utils from "../db/utils"
 import sdk from "../sdk"
+import { authorizedResource } from "./authorized"
 
 export function triggerRowActionAuthorised(
   sourcePath: string,
   actionPath: string
 ) {
-  return async (ctx: Ctx, next: Next) => {
-    // Reusing the existing middleware to extract the value
-    paramSubResource(sourcePath, actionPath)(ctx, () => {})
-    const { resourceId: sourceId, subResourceId: rowActionId } = ctx
+  return async (ctx: UserCtx, next: Next) => {
+    async function getResourceIds() {
+      const sourceId: string = ctx.params[sourcePath]
+      const rowActionId: string = ctx.params[actionPath]
 
-    const isTableId = docIds.isTableId(sourceId)
-    const isViewId = utils.isViewID(sourceId)
-    if (!isTableId && !isViewId) {
-      ctx.throw(400, `'${sourceId}' is not a valid source id`)
+      const isTableId = docIds.isTableId(sourceId)
+      const isViewId = utils.isViewID(sourceId)
+      if (!isTableId && !isViewId) {
+        ctx.throw(400, `'${sourceId}' is not a valid source id`)
+      }
+
+      const tableId = isTableId
+        ? sourceId
+        : utils.extractViewInfoFromID(sourceId).tableId
+      const viewId = isTableId ? undefined : sourceId
+      return { tableId, viewId, rowActionId }
     }
 
-    const tableId = isTableId
-      ? sourceId
-      : utils.extractViewInfoFromID(sourceId).tableId
+    const { tableId, viewId, rowActionId } = await getResourceIds()
 
-    const rowAction = await sdk.rowActions.get(tableId, rowActionId)
+    // Check if the user has permissions to the table/view
+    await authorizedResource(
+      !viewId ? PermissionType.TABLE : PermissionType.VIEW,
+      PermissionLevel.READ,
+      sourcePath
+    )(ctx, () => {})
 
-    if (isTableId && !rowAction.permissions.table.runAllowed) {
+    // Check is the row action can run for the given view/table
+    const rowAction = await sdk.rowActions.get(tableId, rowActionId)
+    if (!viewId && !rowAction.permissions.table.runAllowed) {
       ctx.throw(
         403,
-        `Row action '${rowActionId}' is not enabled for table '${sourceId}'`
+        `Row action '${rowActionId}' is not enabled for table '${tableId}'`
```

Comment:
Row actions are always allowed to be run against the table (assuming you have read access to the table, which I assume the `authorizedResource` takes care of), so I'm just curious what determines if this `table.runAllowed` flag is set or not?

Reply:
I created it in a way that could allow running for tables, but given that the UI does not support it, I will change it to be off by default